File: tessoku/A06 - How Many Guests?/A06.js
const main = (input) => {
    const lines = input.trim().split('\n');
    const [N, Q] = lines[0].split(' ').map(Number);
    const guests = lines[1].split(' ').map(Number);
    const ranges = lines.slice(2).map(line => line.split(' ').map(Number));

    const cumSum = [0];
    for (let i = 0; i < N; i++) {
        cumSum[i + 1] = cumSum[i] + guests[i];
    }

    for (let i = 0; i < Q; i++) {
        const [left, right] = ranges[i];
        console.log(cumSum[right] - cumSum[left - 1]);
    }
}

main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: tessoku/A06 - How Many Guests?/A06.cs
using System;
using System.Linq;

class Program
{
   static void Main()
   {
       int[] firstLine = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
       int N = firstLine[0];
       int Q = firstLine[1];

       int[] guests = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

       long[] cumSum = new long[N + 1];
       for (int i = 0; i < N; i++)
       {
           cumSum[i + 1] = cumSum[i] + guests[i];
       }

       for (int i = 0; i < Q; i++)
       {
           int[] range = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
           int left = range[0];
           int right = range[1];

           Console.WriteLine(cumSum[right] - cumSum[left - 1]);
       }
   }
}